### PR TITLE
chore(docs): Add template docs and workflows

### PR DESCRIPTION
**Pull request by bsmth**
See https://github.com/mdn/mdn/issues/337

### CODE REVIEW

**Review by dipikabh — APPROVED** [review]
Added a question, @bsmth. 
The `dependabot.yml` file seems to be new (was not in other repos). I cannot verify the changes in `dependabot.yml` and `codemirror.js` files. But feel free to merge if you're confident about the changes.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+indent_size = 2
```
[the remark this answers is not in the document]
Comment (bsmth) [reply .editorconfig:8]: Sure thing, added ✅

File: editable-samples/editor/codemirror/lib/codemirror.js [hunk 2]
```diff
@@ -1620,7 +1620,7 @@ function extractLineClasses(type, output) {
     var prop = lineClass[1] ? "bgClass" : "textClass"
     if (output[prop] == null)
       { output[prop] = lineClass[2] }
-    else if (!(new RegExp("(?:^|\s)" + lineClass[2] + "(?:$|\s)")).test(output[prop]))
+    else if (!(new RegExp("(?:^|\\s)" + lineClass[2] + "(?:$|\\s)")).test(output[prop]))
```
Comment (bsmth) [inline editable-samples/editor/codemirror/lib/codemirror.js:1623]: This change is a fix for https://github.com/mdn/js-examples/runs/10603498184

Comment (bsmth) [reply editable-samples/editor/codemirror/lib/codemirror.js:1623]: This is how it's handled in codemirror5:

https://github.com/codemirror/codemirror5/blob/9e864a1bb7c4c452f462d7f8d8be111c8bb8ad6f/src/line/highlight.js#L201

**Review by schalkneethling — APPROVED** [review]
Nice! Immediate benefit by adding CodeQL 🙌

**Comment by bsmth** [comment]
Thanks, Schalk, Dipika 🙌🏻

**Pull request merged**